src/reducers/scrollState.js

```diff
@@ -72,14 +71,18 @@ function scrollState(state = DEFAULT_STATE, action) {
 
       if (newProps.scrollTop && newProps.scrollTop !== oldProps.scrollTop) {
         state = scrollTo(state, newProps.scrollTop);
-        state = updateVisibleRows(state);
       }
 
       if (newProps.scrollToRow && newProps.scrollToRow !== oldProps.scrollToRow) {
         state = scrollToRow(state, newProps.scrollToRow);
-        state = updateVisibleRows(state);
       }
 
+      if (newProps.rowsCount !== state.rowsCount) {
+        state = updateRowCount(state, newProps);
+      }
+
+      state = updateVisibleRows(state);
```

Comment:
No, are tests are severely lacking.  @quixotically is working on adding some more, and if you could help add some tests to keep an eye out for this, that'd be great.  Not sure how you could test this case, but for others maybe mocking updateVisibleRows and ensuring we don't call it when we don't want to.

You can also lean on LDs GridSelectionPerformance and GridScrollPerformance as they'll catch some perf issues introduced here.